src/layouts/root_layout/root_layout_container.js

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {connect} from 'react-redux';
+
+import Config from 'config';
+
+import RootLayout from './root_layout';
+
+function mapStateToProps(state, ownProps) {
+    let locale = Config.DefaultLocale;
+    if (state.entities.users.currentId && state.entities.users.profiles[state.entities.users.currentId]) {
+        locale = state.entities.users.profiles[state.entities.users.currentId].locale;
```

Comment:
try and avoid multiple of this `state.entities.users.profiles[state.entities.users.currentId]`

if you can do something like

```javascript
const users = state.entities.users;
const currentId = users.currentId;
const user = users.profiles[currentId]
```

the differences is not much in terms of performance but it helps a lot when reading